File: telemetry/models/streamable/StreamableRaceInfo.py
from dataclasses import dataclass
from typing import List


@dataclass
class StreamableRaceInfo:
    carClassPosition: int # Streaming (websocket)
    intervalBehindLeader: float  # Streaming (websocket) TODO: Figure out how to get time behind (interval)
    percentageAroundTrack: float  # Streaming (websocket)
    onPitRoad: bool  # Streaming (websocket)
    position: int  # Streaming (websocket)
    onTrackStatus: int  # Streaming (websocket)
    relativeFromCurrentPlayer: float

File: server/src/main/kotlin/models/streamable/StreamableRaceInfo.kt
package models.streamable

data class StreamableRaceInfo(
    val carClassPosition: Int,
    val intervalBehindLeader: Float,
    val percentageAroundTrack: Float,
    val onPitRoad: Boolean,
    val position: Int,
    val onTrackStatus: Int,
    val relativeFromCurrentPlayer: Float
)
